controllers/discovery/initDiscoveryUserData.js

```diff
@@ -21,6 +21,19 @@ const InitDiscoveryUserData = async (req, res) => {
     FROM users A
     WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false`;
 
+    let user_topics = await sequelize.query(
+      `select tp.topic_id from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = :userId`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        }
+      }
+    );
+    let topicIds = user_topics.map((topic) => topic.topic_id);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The query to fetch user topics seems fine, but it's important to handle the case where no topics are associated with a user. In such cases, `topicIds` will be an empty array and this might affect subsequent queries where `topicIds` is used. Consider adding a check for this scenario.
```javascript
let topicIds = user_topics.map((topic) => topic.topic_id);
if (!topicIds.length) {
  // Handle the case when there are no topics associated with the user
}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->